src/the-parser.md

```diff
@@ -1 +1,14 @@
-# The parser
+# The Parser
+
+The parser is responsible for converting raw Rust source code into a structured
+form which is easier for the compiler to work with, usually called an *Abstract
+Syntax Tree*. The bulk of the parser lives in the [libsyntax] crate.
+
+The parsing process is made up of roughly 3 stages,
+
+- lexical analysis - turn a stream of characters into a stream of token trees
+- macro expansion - run `proc-macros` and expand `macro_rules` macros
+- parsing - turn the token trees into an AST
```

Comment:
Here I think I would link to the relevant chapter of the guide -- oh, wait, do we not have a chapter for the AST? We should add one =)

Reply:
I was thinking the AST would fall under the parser section, maybe as a short sub-chapter.